vendor/bat-native-ads/src/bat/ads/internal/ad_events/new_tab_page_ads/new_tab_page_ad_event_viewed.cc

```diff
@@ -34,9 +34,11 @@ AdEventViewed::~AdEventViewed() = default;
 void AdEventViewed::Trigger(
     const NewTabPageAdInfo& ad) {
   database::table::AdEvents database_table(ads_);
+  AdsImpl* ads_copy = ads_;

```

Comment:
Due to `NewTabPAgeAdEventViewed` being short-lived, we need to make a copy of `ads_` otherwise this member variable will not be captured